library/init/category/monad.lean

```diff
@@ -20,57 +20,13 @@ do x, y
 infixl ` >>= `:55 := bind
 infixl ` >> `:55  := has_bind.and_then
 
-section
-set_option auto_param.check_exists false
-
 class monad (m : Type u → Type v) extends applicative m, has_bind m : Type (max (u+1) v) :=
```

Comment:
Is there any reason to keep `has_bind` type class?
It was originally added to make sure we could use the `do` notation for structures that were not monads.
This commit https://github.com/leanprover/lean/commit/cf8dd9e75e4d3b8572677c83244a0f6058e6a992 makes `has_bind` unnecessary, right?